orttraining/orttraining/core/optimizer/memory_optimizer/memory_insight.cc

```diff
@@ -146,13 +146,14 @@ Status GetStashedActivationCandidates(const GraphViewer& graph_viewer,
                            is_forward_nodes);
 
   for (auto& kv : fw_op_output_arg_used_map) {
-    // used by fw and bw, then it is a candidate.
-    if (kv.second.first && kv.second.second) {
-      const Node* n = graph_viewer.GetProducerNode(kv.first);
+    const auto& fw_out_arg = kv.first;
+    const Node* n = graph_viewer.GetProducerNode(fw_out_arg);
+    // Node run in forward pass, and the result is used by bw, then it is a candidate.
+    if (is_forward_nodes.count(n) > 0 && kv.second.second) {
```

Comment:
map.count will return 1 if n exists in the map? and backward node also included in the `is_forward_nodes`?